### PR TITLE
new alert for Observatorium http traffic failures

### PR DESCRIPTION
**Pull request by spaparaju**
This PR adds a new alert for observing Observatorium http traffic failures. This alert is based on metrics collected at `haproxy` router level.

Signed-off-by: spaparaju <paparaju@gmail.com>

### CODE REVIEW

File: observability/prometheusrules.jsonnet [hunk 1]
```diff
@@ -586,3 +588,33 @@ local renderAlerts(name, environment, mixin) = {
...
+        {
+          name: 'observatorium-http-traffic',
+          rules: [
+            {
+              alert: 'ObservatoriumHttpTrafficErrorRateHigh',
+              annotations: {
+                message: 'Observatorium metric queries {{$labels.job}} in {{$labels.namespace}} are failing to handle {{$value | humanize}}% of requests.',
+              },
+              expr: |||
+                ( sum by (namespace, code) (rate(haproxy_backend_http_responses_total{namespace=~"observatorium|telemeter", code!="2xx"} [5m])) / sum by (namespace, code) (rate(haproxy_backend_http_responses_total{namespace=~"observatorium|telemeter", code="2xx"}[5m]))) * 100 > 25
```
[the remark this answers is not in the document]
Comment (spaparaju) [reply observability/prometheusrules.jsonnet]: fixed query based on routes (and independent of underlying namespaces)

File: docs/sop/observatorium.md [hunk 2]
```diff
@@ -88,6 +89,33 @@ Check targets are UP in app-sre Prometheus:
...
+
+`medium`
+
+### Access Required
+
+- Console access to the cluster that runs Observatorium (Currently [telemeter-prod-01 OSD](https://console-openshift-console.apps.telemeter-prod.a5j2.p1.openshiftapps.com/k8s/cluster/projects/observatorium-logs-production))
+- Access to Vault secret for `tenants.yaml` (link for [staging](https://vault.devshift.net/ui/vault/secrets/app-interface/show/app-sre-stage/telemeter-stage/observatorium-observatorium-api); for [production](https://vault.devshift.net/ui/vault/secrets/app-interface/show/app-sre/telemeter-production/observatorium-observatorium-api) you will most likely need to contact [App-SRE](https://coreos.slack.com/archives/CCRND57FW))
+
+### Steps
+- Check the Vault secret for `tenants.yaml` (link for [staging](https://vault.devshift.net/ui/vault/secrets/app-interface/show/app-sre-stage/telemeter-stage/observatorium-observatorium-api); for [production](https://vault.devshift.net/ui/vault/secrets/app-interface/show/app-sre/telemeter-production/observatorium-observatorium-api) for valid Yaml content, valid tenant name, auth information under 'oidc' and 'opa' sections and rate limting information under 'rateLimits' section.
```
[the remark this answers is not in the document]
Comment (spaparaju) [reply docs/sop/observatorium.md]: right, telemeter does not have `opa` bits but MST has `opa` bits. The `tenants.yaml` for both of these tenants are stored under `.../telemeter-stage` at vault. So this specific check is common for both the tenants.

File: docs/sop/observatorium.md [hunk 3]
```diff
@@ -88,6 +89,33 @@ Check targets are UP in app-sre Prometheus:
...
+
+Observatorium API endpoints for metrics are not serving the traffic.
+
+### Severity
+
+`medium`
+
+### Access Required
+
+- Console access to the cluster that runs Observatorium (Currently [telemeter-prod-01 OSD](https://console-openshift-console.apps.telemeter-prod.a5j2.p1.openshiftapps.com/k8s/cluster/projects/observatorium-logs-production))
```
[the remark this answers is not in the document]
Comment (spaparaju) [reply docs/sop/observatorium.md:108]: Changed the URL path to list all the Observatorium projects at the RHOBS cluster. As per the error above, you might not have access to `observatorium-logs-produciton' project

File: resources/observability/prometheusrules/observatorium-http-traffic-stage.prometheusrules.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    prometheus: app-sre
+    role: alert-rules
+  name: observatorium-http-traffic-stage
+spec:
+  groups:
+  - name: observatorium-http-traffic
+    rules:
+    - alert: ObservatoriumHttpTrafficErrorRateHigh
+      annotations:
+        dashboard: https://grafana.app-sre.devshift.net/d/no-dashboard/observatorium-http-traffic?orgId=1&refresh=10s&var-datasource=app-sre-stage-01-prometheus&var-namespace={{$labels.namespace}}&var-job=All&var-pod=All&var-interval=5m
```
[the remark this answers is not in the document]
Comment (spaparaju) [reply resources/observability/prometheusrules/observatorium-http-traffic-stage.prometheusrules.yaml:16]: I see this kinda URLs are common for all the alerts with no-dashboard option. As the name indicates (no-dashboard), these URLs are not functional. I guess we need to fix this across all the alerts.

File: resources/observability/prometheusrules/observatorium-http-traffic-stage.prometheusrules.yaml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        message: Observatorium metric queries {{$labels.job}} in {{$labels.namespace}} are failing to handle {{$value | humanize}}% of requests.
+        runbook: https://github.com/rhobs/configuration/blob/main/docs/sop/observatorium.md#observatoriumhttptrafficerrorratehigh
```
[the remark this answers is not in the document]
Comment (spaparaju) [reply resources/observability/prometheusrules/observatorium-http-traffic-stage.prometheusrules.yaml:18]: It URL will be operational once this PR gets merged.

File: observability/prometheusrules.jsonnet (hunk 1 above)
[the remark this answers is not in the document]
Comment (spaparaju) [reply observability/prometheusrules.jsonnet]: All the auth errors result in 4xxs, so they are useful information to capture. Fixed query to capture both 4xx and 5xx http error codes.

File: observability/prometheusrules.jsonnet [hunk 6]
```diff
@@ -586,3 +588,33 @@ local renderAlerts(name, environment, mixin) = {
...
+    // Alerts for HTTP traffic.
+    prometheusAlerts+:: {
+      groups: [
+        {
+          name: 'observatorium-http-traffic',
+          rules: [
+            {
+              alert: 'ObservatoriumHttpTrafficErrorRateHigh',
+              annotations: {
+                message: 'Observatorium metric queries {{$labels.job}} in {{$labels.namespace}} are failing to handle {{$value | humanize}}% of requests.',
```
[the remark this answers is not in the document]
Comment (spaparaju) [reply observability/prometheusrules.jsonnet]: good catch. fixed the query to remove not-applicable labels.

File: observability/prometheusrules.jsonnet [hunk 7]
```diff
@@ -586,3 +588,33 @@ local renderAlerts(name, environment, mixin) = {
...
+              },
+              expr: |||
+                ( sum by (route) (rate(haproxy_backend_http_responses_total{route=~"observatorium.*|telemeter.*|infogw.*", code!="2xx"} [5m])) / sum by (route) (rate(haproxy_backend_http_responses_total{route=~"observatorium.*|telemeter.*|infogw.*", code="2xx"}[5m]))) * 100 > 25
```
[the remark this answers is not in the document]
Comment (spaparaju) [reply observability/prometheusrules.jsonnet]: Fixed query to capture both 4xx and 5xx http error codes.

File: resources/observability/prometheusrules/observatorium-http-traffic-stage.prometheusrules.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  groups:
+  - name: observatorium-http-traffic
+    rules:
+    - alert: ObservatoriumHttpTrafficErrorRateHigh
+      annotations:
+        dashboard: https://grafana.app-sre.devshift.net/d/no-dashboard/observatorium-http-traffic?orgId=1&refresh=10s&var-datasource=app-sre-stage-01-prometheus&var-namespace={{$labels.namespace}}&var-job=All&var-pod=All&var-interval=5m
+        message: Observatorium route  {{$labels.route}}  are failing to handle {{$value | humanize}}% of requests.
+        runbook: https://github.com/rhobs/configuration/blob/main/docs/sop/observatorium.md#observatoriumhttptrafficerrorratehigh
+      expr: |
+        ( sum by (route) (rate(haproxy_backend_http_responses_total{route=~"observatorium.*|telemeter.*|infogw.*", code="4xx|5xx"} [5m])) / sum by (route) (rate(haproxy_backend_http_responses_total{route=~"observatorium.*|telemeter.*|infogw.*"}[5m]))) * 100 > 25
```
Comment (ianbillett) [inline resources/observability/prometheusrules/observatorium-http-traffic-stage.prometheusrules.yaml:20]: Discussed during team sync - `4xx` is a client side error i.e. the client is using the service wrong, `5xx` is a server side error. For this alert - we should only be capturing `5xx` error codes.

File: observability/prometheusrules.jsonnet (hunk 7 above)
[the remark this answers is not in the document]
Comment (spaparaju) [reply observability/prometheusrules.jsonnet]: update: Fixed query to capture  only the 5xx http error codes.

**Review by matej-g — COMMENTED** [review]
Looking good! :muscle: Just one last comment.

**Review by ianbillett — APPROVED** [review]

**Review by matej-g — APPROVED** [review]
:rocket:

**Pull request merged**